### PR TITLE
Added PackageConfigHelpers to make the library easier to use

### PR DESCRIPTION
**Pull request by doug1234**

### CODE REVIEW

**Comment by doug1234** [comment]
Any idea what is going wrong with the build here.  I built windows and linux on my own and it worked but it is failing on ubuntu and macos here in the install phase for reasons I can't understand.

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,7 +80,29 @@ target_link_libraries(OpenDDW
 
 OPENDDS_TARGET_SOURCES(OpenDDW idl/std_qos.idl)
 
+INCLUDE(CMakePackageConfigHelpers)
+
+INSTALL(EXPORT ${PROJECT_NAME}
+	NAMESPACE ${PROJECT_TARGET_PREFIX}::
+	FILE ${PROJECT_TARGET_FILE}
+	DESTINATION ${CONFIG_FILE_INSTALL_LOCATION}
+)
```
Comment (iguessthislldo) [inline CMakeLists.txt:107]: I think the issue is there are no targets in `OpenDDW` to export. Looking at my CMake work in OpenDDS as a reference, I'm calling [this in a helper function as I'm defining targets](https://github.com/iguessthislldo/OpenDDS/blob/410c7a45563904d4796b276ba215dadfdfaf8208/cmake/opendds_build_helpers.cmake#L63) which is adding the targets to `opendds_targets`. After all the targets are defined I'm calling `install(EXPORT opendds_targets ...)` [here](https://github.com/iguessthislldo/OpenDDS/blob/410c7a45563904d4796b276ba215dadfdfaf8208/CMakeLists.txt#L84-L87) in the root CMakeLists. So I think you need a [`install(TARGETS OpenDDW ...)`](https://cmake.org/cmake/help/latest/command/install.html#install-targets) before the `install(EXPORT)`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -56,7 +71,7 @@ set(OPENDDS_TARGETS
   OpenDDS::Rtps_Udp
 )
 
-target_include_directories(OpenDDW INTERFACE src)
+#target_include_directories(OpenDDW INTERFACE src)
```
Comment (iguessthislldo) [inline CMakeLists.txt]: Probably need a `BUILD_INTERFACE`/`INSTALL_INTERFACE` separation here. See [`target_include_directories`](https://cmake.org/cmake/help/latest/command/target_include_directories.html), above the "Creating Relocatable Packages" section.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (doug1234) [reply CMakeLists.txt:107]: Thanks.  I made the change you suggested and found some other stuff to fix/improve.  Odd that it wasn't failing on windows...

File: CMakeLists.txt (hunk 2 above)
Comment (doug1234) [reply CMakeLists.txt]: Should be good now.

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**